lang/py/avro/name.py

```diff
@@ -91,6 +93,10 @@ def __eq__(self, other: object) -> bool:
         """Equality of names is defined on the fullname and is case-sensitive."""
         return hasattr(other, "fullname") and self.fullname == getattr(other, "fullname")
 
+    @property
+    def name(self) -> Optional[str]:
```

Comment:
Let's add some unit tests for this new method with inputs like:
1) "Record"
2) "ns.Record"
3) "longer.ns.Record"
4) ".Record"

Reply:
Added some.